src/amo/components/CollectionDetails/index.js

```diff
@@ -0,0 +1,90 @@
+/* @flow */
+import * as React from 'react';
+import { compose } from 'redux';
+
+import { convertFiltersToQueryParams } from 'amo/reducers/collections';
+import translate from 'core/i18n/translate';
+import { sanitizeHTML } from 'core/utils';
+import Button from 'ui/components/Button';
+import LoadingText from 'ui/components/LoadingText';
+import MetadataCard from 'ui/components/MetadataCard';
+import type {
+  CollectionFilters,
+  CollectionType,
+} from 'amo/reducers/collections';
+import type { I18nType } from 'core/types/i18n';
+
+import './styles.scss';
+
+export type Props = {|
+  collection: CollectionType | null,
+  editUrl: string,
```

Comment:
Requiring `editUrl` puts an unnecessary burden on the caller. `CollectionDetail` should be able to create an edit URL with just the `collection` object.

We've struggled with how to do this in the past. I think the best way is to introduce two new functions in `reducers/collections`: `collectionUrl()` and `editCollectionUrl()` (or named whatever).

There are a couple scenarios in `Collection` where it needs to create a URL without having a `collection` object but the function could allow for that by accepting `authorUsername` and `slug` values.

Reply:
I agree, and will add the functions to this PR.